src/main/java/com/microsoft/sqlserver/jdbc/SqlVariant.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc;
+
+import java.text.MessageFormat;
+
+/**
+ * This class holds information regarding the basetype of a sql_variant data.
+ *
+ */
+
+/**
+ * Enum for valid probBytes for different TDSTypes
+ *
+ */
+enum SqlVariant_ProbBytes {
+    INTN(0),
+    INT8(0),
+    INT4(0),
+    INT2(0),
+    INT1(0),
+    FLOAT4(0),
+    FLOAT8(0),
+    DATETIME4(0),
+    DATETIME8(0),
+    MONEY4(0),
+    MONEY8(0),
+    BITN(0),
+    GUID(0),
+    DATEN(0),
+    TIMEN(1),
+    DATETIME2N(1),
+    DECIMALN(2),
+    NUMERICN(2),
+    BIGBINARY(2),
+    BIGVARBINARY(2),
+    BIGCHAR(7),
+    BIGVARCHAR(7),
+    NCHAR(7),
+    NVARCHAR(7);
+
+    private final int intValue;
+
+    private static final int MAXELEMENTS = 23;
+    private static final SqlVariant_ProbBytes valuesTypes[] = new SqlVariant_ProbBytes[MAXELEMENTS];
+
+    int intValue() {
+        return intValue;
+    }
+
+    private SqlVariant_ProbBytes(int intValue) {
+        this.intValue = intValue;
+    }
+
```

Comment:
let's move private SqlVariant_ProbBytes(int intValue) up since it's the constructor for the enum. Also change the name of intValue() to getIntValue() ?